### PR TITLE
fix: Fix issue with applying transformation to bevelled cylinders

### PR DESCRIPTION
**Pull request by noraemi**
Applying transforms to the cylinder did not work as intended. These fixes should place the side disks at the right place as well as fixing the visualization of the cylinder length.

### CODE REVIEW

File: Core/src/Surfaces/CylinderBounds.cpp [hunk 1]
```diff
@@ -167,7 +167,11 @@ std::vector<Acts::Vector3> Acts::CylinderBounds::createCircles(
   if ((bevelMinZ != 0. || bevelMaxZ != 0.) && vertices.size() % 2 == 0) {
     auto halfWay = vertices.end() - vertices.size() / 2;
     double mult{1};
-    auto func = [&mult](Vector3& v) { v(2) += v(1) * mult; };
+    auto func = [&mult, &ctrans](Vector3& v) {
+      v = ctrans.inverse() * v;
```
Comment (paulgessinger) [inline Core/src/Surfaces/CylinderBounds.cpp:171]: Consider caching the inverse here and capturing it alongside `ctrans`.

Comment (noraemi) [reply Core/src/Surfaces/CylinderBounds.cpp]: fixed

File: Core/src/Surfaces/CylinderBounds.cpp [hunk 2]
```diff
@@ -167,7 +167,12 @@ std::vector<Acts::Vector3> Acts::CylinderBounds::createCircles(
   if ((bevelMinZ != 0. || bevelMaxZ != 0.) && vertices.size() % 2 == 0) {
     auto halfWay = vertices.end() - vertices.size() / 2;
     double mult{1};
-    auto func = [&mult](Vector3& v) { v(2) += v(1) * mult; };
+    auto invCtrans = ctrans.inverse();
```
Comment (paulgessinger) [inline Core/src/Surfaces/CylinderBounds.cpp:170]: Sorry to comment again, but I think if you use `auto` here, it won't actually do the inversion here (might be wrong). You can force this by either using a specific type or using `inverse().eval()`.

Comment (noraemi) [reply Core/src/Surfaces/CylinderBounds.cpp:170]: Seems to work for me, at least dumping a cylinder (or even multiple) gives the expected result
image

**Review by paulgessinger — APPROVED** [review]

**Pull request merged**